firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/HAL/Interfaces/Endian.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ *      Author: Ethan Li
+ *
+ *  Endianness conversions between host and network order.
+ *  STM32 is little-endian, while network order is big-endian, so
+ *  the conversion functions reverse the order of bytes.
+ *
+ *  NOTE: this implementation relies on builtins provided by GCC/Clang.
+ *  You will need to compile using a relatively modern version of either
+ *  compiler.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace Pufferfish::HAL::Interfaces {
+
+/**
+ * Convert endianness of a 64-bit data in host-order to network-order
+ * @param hostValue a 64-bit number in host-order
+ * @return the given number in network-order
+ */
+constexpr uint64_t hton(uint64_t host_value) noexcept {
+  return __builtin_bswap64(host_value);
```

Comment:
Would it be possible for `HAL/Interfaces/Endian.h` to only provide forward declarations of these `constexpr` functions, and then for `HAL/STM32/Endian.h` to provide the definitions of these functions? i.e. we copy this file into `HAL/STM32`, add it as an `#include` into `HAL/STM32/HAL.h`, and then delete the implementations from this file. If this is possible, we would also need to ensure that `main.cpp` includes `HAL/STM32/HAL.h` in front of the includes of Driver and Util files.